mne/minimum_norm/resolution_matrix.py

```diff
@@ -0,0 +1,175 @@
+"""Compute resolution matrix for linear estimators."""
+from copy import deepcopy
+
+import numpy as np
+
+from mne import pick_channels_forward, EvokedArray
+from mne.io.constants import FIFF
+from mne.utils import logger
+from mne.forward.forward import is_fixed_orient, convert_forward_solution
+
+from mne.minimum_norm import apply_inverse
+
+
+def make_resolution_matrix(forward, inverse_operator, method='dSPM',
+                           lambda2=1. / 9.):
+    """Compute resolution matrix for linear inverse operator.
+
+    Parameters
+    ----------
+    forward: dict
+        Forward Operator.
+    inverse_operator: Instance of InverseOperator
+        Inverse operator.
+    method: str
+        Inverse method to use (MNE, dSPM, sLORETA).
+    lambda2: float
+        The regularisation parameter.
+
+    Returns
+    -------
+    resmat: array, shape (n_dipoles, n_dipoles)
+        Resolution matrix (inverse operator times forward operator).
+    """
+    # make sure forward and inverse operator match
+    inv = inverse_operator
+    fwd = _convert_forward_match_inv(forward, inv)
+
+    # don't include bad channels
+    # only use good channels from inverse operator
+    bads_inv = inv['info']['bads']
+
+    # good channels
+    ch_names = [c for c in inv['info']['ch_names'] if (c not in bads_inv)]
+
+    fwd = pick_channels_forward(fwd, ch_names, ordered=True)
+
+    # get leadfield matrix from forward solution
+    leadfield = fwd['sol']['data']
+
+    invmat = _get_matrix_from_inverse_operator(inv, fwd,
+                                               method=method, lambda2=lambda2)
+
+    resmat = invmat.dot(leadfield)
+
+    logger.info('Dimensions of resolution matrix: %d by %d.' % resmat.shape)
+
+    return resmat
+
+
+def _convert_forward_match_inv(fwd, inv):
+    """Helper to ensure forward and inverse operators match."""
+    # did inverse operator use fixed orientation?
+    is_fixed_inv = inv['eigen_leads']['data'].shape[0] == inv['nsource']
```

Comment:
same here use:

inverse_operator['source_ori'] != FIFF.FIFFV_MNE_FREE_ORI

or better make a private function you can use at the 2 places